calligator/boards.py

```diff
@@ -422,26 +430,58 @@ def fill_points(self, corners, ids=None):
             return out
 
     def detect_image(self, image, subpix=True):
+
+        ids = None
+
         if len(image.shape) == 3:
             gray = cv2.cvtColor(image, cv2.COLOR_BGR2GRAY)
         else:
             gray = image
 
         size = self.get_size()
-        ret, corners = cv2.findChessboardCorners(gray, size,
-                                                 self.DETECT_PARAMS)
-
-        if ret and subpix:
-            corners = cv2.cornerSubPix(gray, corners, (3, 3), (-1, -1),
-                                       self.SUBPIX_CRITERIA)
+        pattern_found, corners = cv2.findChessboardCorners(gray, size, self.DETECT_PARAMS)
 
-        if corners is None:
-            ids = None
-        else:
+        # Does <findChessBoardCorners()> believe that it found the pattern?
+        if pattern_found:
             ids = self.ids
+            # Do we want sub-pixel accuracy?
+            if subpix:
+                # TODO: The docs at (https://docs.opencv.org/3.4.0/dd/d1a/group__imgproc__feature.html
+                #  #ga354e0d7c86d0d9da75de9b9701a9a87e) do not specify what the return value of <cornerSubPix()>
+                #  will be if the function fails. Undefined behavior from my perspective. Could break
+                #  things if <cornerSubPix()> fails.
+                corners = cv2.cornerSubPix(gray, corners, (3, 3), (-1, -1), self.SUBPIX_CRITERIA)
+
+        # Does a human want to manually verify the detection?
+        if self.manually_verify:
+            # <manually_verify_board_detection()> returns True if a human selects that the detection is correct.
+            # Otherwise, it returns false.
+            if self.manually_verify_board_detection(gray, corners, pattern_found):
+                pass
+            else:
+                corners = None
+                ids = None
 
         return corners, ids
 
+    def manually_verify_board_detection(self, image, corners, pattern_was_found):
```

Comment:
This should not even be called when pattern is not found, so the `pattern_was_found` argument is redundant.